pyscf/lib/libqcschema.py

```
@@ -0,0 +1,256 @@
+        geo = np.array(qcschema_dict["trajectory"][step]["molecule"]["geometry"])    
+
+    if(to_Angstrom):
+        # convert Bohr to Angstrom
+        geo = geo*BOHR
+
+    NAtoms = len(syms)
+    geo = np.reshape(geo, (NAtoms,3))
+
+    PySCF_atoms = list(zip(syms, geo))
+
+    # Return as string or return as xyz-format string (i.e. top is NAtoms,blankline)
+    if(xyz):
+        bldstr = f'{NAtoms}\n\n'
+        for element, coordinates in PySCF_atoms:
+            bldstr += f'{element} {coordinates[0]}, {coordinates[1]}, {coordinates[2]}\n'
+            PySCF_atoms = bldstr
+    return PySCF_atoms
+
+def load_qcschema_hessian(qcschema_dict):
+    '''
+    Does: loads hessian from qcschema format dictionary
+    Input:
+        qcschema_dict
+
+    Returns: hessian with format (N,N,3,3)
+    '''
+    # qcschema_dict: pass in dict that has the qcschema output json loaded into it
+
+    # load qcschema hessian
+    qc_h = []
+    qc_h = qcschema_dict["return_result"]
+
+    # Get Number of atoms N
+    syms = np.array(qcschema_dict["molecule"]["symbols"])
+    NAtom = len(syms)
+
+    # reshape from (3N)**2 array to (N,N,3,3)
+    hessian = np.array(qc_h).reshape(NAtom,NAtom,3,3)
+    return hessian
+
+def load_qcschema_scf_info(qcschema_dict):
+    '''
+    Does: loads scf info from qcschema format dictionary
+    Input:
+        qcschema_dict
+
+    Returns:
+        scf_dict: contains the relevent scf info only
+    '''
+    
+    # Restricted wfn has schema scf_occupations_a occ of 1 or 0.
+    # Need to double if RHF/RKS/ROHF
+    method = qcschema_dict["keywords"]["scf"]["method"]
+    if(method == 'rks' or method == 'roks' or method == 'rhf' or method == 'rohf'):
+        OccFactor = 2.0
+        have_beta = False
+    elif(method == 'uks' or method == 'uhf'):
+        OccFactor = 1.0
+        have_beta = True
+    elif(method == 'gks' or method == 'ghf'):
+        OccFactor = 1.0
+        have_beta = False
```

please raise an exception for other cases